clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -469,6 +474,8 @@ public StoreIngestionTask(
         this::pauseConsumption,
         this::resumeConsumption);
     this.storeRepository.registerStoreDataChangedListener(this.storageUtilizationManager);
+    this.storeVersionRoleChangedListener = new StoreVersionRoleChangedListener(this, store, versionNumber);
+    this.storeRepository.registerStoreDataChangedListener(storeVersionRoleChangedListener);
```

Comment:
I don't think this is the correct way to use ZK callback. Any store change will trigger callbacks in all registered listeners, meaning that changes in Store A will trigger listener for store B's SIT with the current implementation; and current implementation will end up adding too many listeners.

I believe the correct way is to build only one listener in ingestion service, when it receives changes for a specific store, invoke version change API in its SITs.

Reply:
Another way would be let `SIT` thread periodically check the version status and if there is a change, invoke version status change API.
The subscription change is not time sensitive, so it doesn't have to react to ZK change immediately.

Reply:
I think let SIT thread periodically check the version status would simplify the logic. It will not incur more threads and simplify the logic, and probably we do not need to enqueue consumer action.